packages/xstate-test/test/index.test.ts

```diff
@@ -114,9 +115,10 @@ describe('events', () => {
     const testedEvents: any[] = [];
 
     const testModel = createTestModel(testMachine, {
-      eventCases: {
-        EVENT: (state) => state.context.values.map((value) => ({ value }))
-      }
+      events: (state) =>
+        state.context.values.map(
+          (value) => ({ type: 'EVENT', value } as EventFrom<typeof testMachine>)
+        )
```

Comment:
This is closer to what I would expect from an average user:
```suggestion
        state.context.values.map((value) => ({ type: 'EVENT', value } as const))
```

Reply:
@Andarist Ideally we shouldn't have to specify `as const` or anything in there; do you know why inference is lacking here?

Reply:
Not (yet). We suffer from this problem in multiple places. The problem arises when the return type of the expression should be constrained by what we want to infer from other sources: [TS playground](https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEAzArgOzAFwJYHtXwxAGcMAeAFQFEA3EVDeEAD0NWCPgG8CBPABxAAueCRhZUAc3gBfAHwAKHACMAVsM4AoePAkgMNOhiLD5ASngBeWfDhRgeCD3hVa9ANoBdANwbpp4dQ4WMA+GgD0YQQAFlgcAO44MADWHPKoOPDiiCBw6CCmGoQkpNwY-ELwAEQAYgDytZUyCprauvquRibmVvBupeXCNfWN0h4ANL6moaCQsAizRBwAslBgMaggFAb01i0EMFCoRFjYeCYg1MIuht3WgcE+0hrhkRgxHPbEqADkDAnJ8HkWRydDA+Q0M2gcCQaEwuHwRQwACYth1GCw6OwuLwBMJROIpHJ5Fp4ABbVbrCorNbiTbXHYTbTKNRcEmtPTbTqA242EB2BxOekYTw+bRPfzwe4hZ6IpHyTjSeBQZYU2klHEVIYNJpjVns9qGYzcyzWPoawZ1bWjCZ+HxAA)